allennlp/models/multitask.py

```diff
@@ -0,0 +1,109 @@
+from collections import defaultdict
+import inspect
+from typing import Dict, List
+
+from overrides import overrides
+import torch
+
+from allennlp.data import Vocabulary
+from allennlp.modules import Backbone
+from allennlp.models.model import Model
+from allennlp.models.heads import Head
+from allennlp.nn import InitializerApplicator
+
+
+
+# TODO:
+# 1. implement get_metrics
+# 2. implement make_output_human_readable
+# 3. implement selective arguments / which heads to run at training vs. eval, etc.
+
+
+def get_forward_arguments(module: torch.nn.Module) -> List[str]:
+    signature = inspect.signature(module.forward)
+    arguments = []
+    for arg in signature.parameters:
+        if arg == "self":
+            continue
+        arguments.append(arg)
+    return arguments
+
+
+class MultiTaskModel(Model):
+    """
+    initializer : `InitializerApplicator`, optional (default=`InitializerApplicator()`)
+        If provided, will be used to initialize the model parameters.
```

Comment:
This will need some more documentation.

Reply:
Why?  Oh, this reminds me that I need to make `self._heads` into a `ModuleDict`, so that it trains correctly.

Ok, I thought you were talking about the initializer parameter itself, then I went and looked at the code to fix the `ModuleDict` error and realized you were probably talking about the whole docstring in general.  Yes, definitely.